go-controller/pkg/ovn/ovn.go

```diff
@@ -1179,3 +1196,107 @@ func shouldUpdate(node, oldNode *kapi.Node) (bool, error) {
 
 	return true, nil
 }
+
+// checkDBMembers checks if the db-ids of the db instances have changed.
+// In case all the db-ids have departed from the original list, we treat
+// this as a need for the master to re-sync all the dbs. This is particularly
+// relevant for when the dbs get corrupted for some unknown reason.
+// This kill-switch lets us recover and resync the dbs to known state.
+func (oc *Controller) checkDBMembers(dbName string) {
+	var appCtl func(args ...string) (string, string, error)
+	var dbIds []string
+	var numDBIdsChanged int
+	if dbName == "OVN_Northbound" {
+		appCtl = util.RunOVNNBAppCtl
+		dbIds = oc.ovnNBDBIds
+	} else {
+		appCtl = util.RunOVNSBAppCtl
+		dbIds = oc.ovnSBDBIds
+	}
+
+	out, stderr, err := appCtl("cluster/status", dbName)
+	if err != nil {
+		klog.Warningf("Unable to get cluster status for: %s, stderr: %v, err: %v", dbName, stderr, err)
+		return
+	}
+	r, _ := regexp.Compile(`([a-z0-9]{4}) at ((ssl|tcp):\[?[a-z0-9.:]+\]?)`)
+	members := r.FindAllStringSubmatch(out, -1)
+	// Currently the dbIds slice is never "updated".
```

Comment:
why not?

Reply:
see https://github.com/ovn-org/ovn-kubernetes/pull/1717#discussion_r492356775